framework/PostProcessors/FTstructure.py

```diff
@@ -0,0 +1,160 @@
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Created on April 30, 2018
+
+@author: mandd
+"""
+
+from __future__ import division, print_function , unicode_literals, absolute_import
+import warnings
+warnings.simplefilter('default', DeprecationWarning)
+
+#Internal Modules---------------------------------------------------------------
+import MessageHandler
+from utils import utils
+from .FTgate import FTgate
+#Internal Modules End-----------------------------------------------------------
+
+#External Modules---------------------------------------------------------------
+import numpy as np
+import xml.etree.ElementTree as ET
+import copy
+import itertools
+from collections import OrderedDict
+#External Modules End-----------------------------------------------------------
+
+class FTstructure():
+  """
+    This is the base class of the FT structure which actually handles FT structures which is used by the FTimporter and the FTmodel
+  """
+  def __init__(self, inputs, topEventID):
+    """
+      This method executes the postprocessor action.
+      @ In,  inputs, list, list of file objects
+      @ Out, outcome, dict, dict containing the processed FT
+    """
+    self.basicEvents = []
+    self.houseEvents = {}
+    self.gateList    = {}
+    self.gateID      = []
+    self.topEventID  = topEventID
+
+    for file in inputs:
+      faultTree = ET.parse(file.getPath() + file.getFilename())
+      faultTree = findAllRecursive(faultTree,'opsa-mef')
+
+      for gate in findAllRecursive(faultTree[0], 'define-gate'):
+        FTGate = FTgate(gate)
+        self.gateList[gate.get('name')] = FTGate
+        self.gateID.append(gate.get('name'))
+
+      for basicEvent in findAllRecursive(faultTree[0], 'basic-event'):
+        self.basicEvents.append(basicEvent.get('name'))
+
+      for houseEvent in findAllRecursive(faultTree[0], 'define-house-event'):
+        value = houseEvent.find('constant').get('value')
+        if value in ['True','true']:
+          value = 1.
+        elif value in ['False','false']:
+          value = 0.
+        else:
+          raise IOError('FTImporterPostProcessor Post-Processor ' + self.name + ': house event ' + str(basicEvent.get('name')) + ' has a not boolean value (True or False)')
+        self.houseEvents[houseEvent.get('name')] = value
+
+    if not self.topEventID in self.gateID:
+      raise IOError('FTImporterPostProcessor: specified top event ' + str(self.topEventID) + ' is not contained in the fault-tree; available gates are: ' + str(self.gateID))
+
+  def returnDict(self):
+    """
+      This method returns
```

Comment:
what?

Reply:
also the name of the method is not really "accurate" for what this method is returning